### PR TITLE
Remove dependency with Rx

### PR DESCRIPTION
**Pull request by juankysoriano**
Given the usage of Rx in this library is almost anecdotical let's don't force others to do have to include those dependencies on their projects if they are using the library.

This relates to this issue: https://github.com/novoda/simplechromecustomtabs/issues/10

Note: This increases build tools and support libraries version number.

### CODE REVIEW

**Comment by ouchadam** [comment]
:100: seems like RX wasn't really needed after all!

File: library/src/test/java/com/novoda/simplechromecustomtabs/provider/SimpleChromeCustomTabsAvailableAppProviderTest.java [hunk 1]
```diff
@@ -110,4 +106,13 @@ private void givenThatPackageIsNotFound() {
         givenThatPackageIsEmpty();
     }
 
+    private static class StubExecutor implements Executor {
+        private final Handler mHandler = new Handler(Looper.getMainLooper());
+
+        @Override
+        public void execute(Runnable command) {
+            mHandler.post(command);
```
Comment (amlcurran) [inline library/src/test/java/com/novoda/simplechromecustomtabs/provider/SimpleChromeCustomTabsAvailableAppProviderTest.java]: as the test is run on the main thread by default you could just do `command.run()` here

Comment (juankysoriano) [inline library/src/test/java/com/novoda/simplechromecustomtabs/provider/SimpleChromeCustomTabsAvailableAppProviderTest.java]: you are right @amlcurran

Comment (ouchadam) [inline library/src/test/java/com/novoda/simplechromecustomtabs/provider/SimpleChromeCustomTabsAvailableAppProviderTest.java]: I thought they were on a `Test` thread?

**Comment by juankysoriano** [comment]
@ouchadam I included it given the hype I had with Rx when I started this library, which is also when I started with novoda.

**Comment by fourlastor** [comment]
:+1: :ship:

File: library/src/main/java/com/novoda/simplechromecustomtabs/provider/SimpleChromeCustomTabsAvailableAppProvider.java [hunk 2]
```diff
@@ -4,66 +4,37 @@
...
 
 public class SimpleChromeCustomTabsAvailableAppProvider implements AvailableAppProvider {
 
-    private final Observable<String> observable;
     private final BestPackageFinder bestPackageFinder;
+    private final Executor executor;
 
-    SimpleChromeCustomTabsAvailableAppProvider(Observable<String> observable, BestPackageFinder bestPackageFinder) {
-        this.observable = observable;
+    SimpleChromeCustomTabsAvailableAppProvider(BestPackageFinder bestPackageFinder, Executor executor) {
```
Comment (mr-archano) [inline library/src/main/java/com/novoda/simplechromecustomtabs/provider/SimpleChromeCustomTabsAvailableAppProvider.java]: :100: well done @juankysoriano!

**Comment by alosdev** [comment]
:+1:

**Pull request merged**